### PR TITLE
Data Types Chapter

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: content/data-types/_index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Learning Objectives
+
+After completing everything in this chapter, you should be able to do the following:
```
Comment (codinglikeagirl42) [inline content/data-types/_index.md:17]:
```suggestion
After completing this chapter, you should be able to do the following:
```

File: content/data-types/_index.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+1. `Console.WriteLine`
+1. `Console.ReadLine`
+1. comments
+1. `char`
+1. array
+1. object
+1. method
+1. static
+1. public
+
```
Comment (codinglikeagirl42) [inline content/data-types/_index.md:50]: Please reformat based on last weeks conversation.

File: content/data-types/exercises/_index.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+{{% expand "Click Here for Answer" %}}
+
+   ```csharp {linenos = table}
+      Console.WriteLine("What is your name?");
+      string myName = Console.ReadLine();
+      Console.WriteLine("Hello " + myName + "!");
+   ```
+
+{{% /expand %}}
```
Comment (codinglikeagirl42) [inline content/data-types/exercises/_index.md:60]:
```suggestion
  {{% expand "Check your solution" %}}
  
     ```csharp {linenos = table}
        Console.WriteLine("What is your name?");
        string myName = Console.ReadLine();
        Console.WriteLine("Hello " + myName + "!");
     ```
  
  {{% /expand %}}
```

File: content/data-types/exercises/_index.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      }
+      else 
+      {
+         Console.WriteLine("false");
+      }
+   ```
+
+   {{% /expand %}}
+
+1. Extend the previous exercise. Assume the user enters a word that is in the sentence. Print out its index within the string and its length. Next, remove the word from the string and print the sentence again to confirm your code. Remember that strings are *immutable*, so you will need to reassign the old sentence variable or create a new one to store the updated phrase.
```
Comment (codinglikeagirl42) [inline content/data-types/exercises/_index.md:170]: Noticed that if you use Alice it only removes the first instance.  I'm assuming they do not have the knowledge yet to correct for this?  If they do could this be reworked? Or if not, add a note?

**Review by codinglikeagirl42 — CHANGES_REQUESTED** [review]
Please let me know if you have any questions.

File: content/data-types/exercises/_index.md (hunk 4 above)
Comment (gildedgardenia) [reply content/data-types/exercises/_index.md:170]: That is expected and we do not need to add a note for students as they have learned about the logic of removing one instance and multiple instances of substrings in Unit 1.

File: content/data-types/exercises/_index.md (hunk 3 above)
Comment (gildedgardenia) [reply content/data-types/exercises/_index.md:60]: Done!

File: content/data-types/_index.md (hunk 2 above)
Comment (gildedgardenia) [reply content/data-types/_index.md:50]: Done!

File: content/data-types/exercises/_index.md (hunk 3 above)
Comment (codinglikeagirl42) [reply content/data-types/exercises/_index.md]: Thanks

File: content/data-types/_index.md (hunk 2 above)
Comment (codinglikeagirl42) [reply content/data-types/_index.md]: There are two styles with regards to the formatting for Key Terminology.

Style 1 -> Data Types, Classes and Unit Testing
Style 2 -> Control Flow & Collections and Classes Part 2

Suggest on deciding on one style to make the book feel more cohesive.

**Review by codinglikeagirl42 — APPROVED** [review]
To remove the already committed .DS_Store files from the repo one can run the following command in their terminal:

find . -name .DS_Store -print0 | xargs -0 git rm -f --ignore-unmatch

Commit and push up.

Left one other suggestion for you to look at, then merge away!

File: content/data-types/_index.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+---
+title: "Data Types"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 2
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/_index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/next-steps.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+title: "Next Steps"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 4
+originalAuthor: <no value> # to be set by page creator
+originalAuthorGitHub: <no value> # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/next-steps.md:9]: Please add metadata before merging

File: content/data-types/reading/_index.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Reading"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 1
+originalAuthor: <no value> # to be set by page creator
+originalAuthorGitHub: <no value> # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/reading/_index.md:9]: Please add metadata before merging

File: content/data-types/reading/data-types/index.md [hunk 8]
```diff
@@ -0,0 +1,291 @@
+---
+title: "Data Types"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 1
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/reading/data-types/index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/exercises/_index.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
+---
+title: "Exercises"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 2
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/exercises/_index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/reading/objects-primer/index.md [hunk 10]
```diff
@@ -0,0 +1,252 @@
+---
+title: "Objects and Methods, a Primer"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 4
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/reading/objects-primer/index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/reading/some-csharp-practice/index.md [hunk 11]
```diff
@@ -0,0 +1,243 @@
+---
+title: "Some C# Practice"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 2
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/reading/some-csharp-practice/index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/reading/strings-chars-and-arrays/index.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
+---
+title: "Strings, Characters, and Arrays"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 3
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/reading/strings-chars-and-arrays/index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/studio/_index.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
+---
+title: "Studio: Area of a Circle"
+date: 2023-01-17T16:28:22-06:00
+draft: false
+weight: 3
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/data-types/studio/_index.md:9]: Please add my information to the MetaData before merging.

File: content/data-types/_index.md (hunk 2 above)
Comment (gildedgardenia) [reply content/data-types/_index.md]: We will do that! Thanks! Merging for now

**Pull request merged**